servers/physics_3d/body_3d_sw.cpp

```diff
@@ -501,20 +501,17 @@ void Body3DSW::integrate_forces(real_t p_step) {
 
 	if (mode == PhysicsServer3D::BODY_MODE_KINEMATIC) {
 		//compute motion, angular and etc. velocities from prev transform
-		linear_velocity = (new_transform.origin - get_transform().origin) / p_step;
+		motion = new_transform.origin - get_transform().origin;
+		do_motion = true;
+		linear_velocity = motion / p_step;
 
 		//compute a FAKE angular velocity, not so easy
-		Basis rot = new_transform.basis.orthonormalized().transposed() * get_transform().basis.orthonormalized();
+		Basis rot = new_transform.basis.orthonormalized() * get_transform().basis.orthonormalized().transposed();
 		Vector3 axis;
 		real_t angle;
 
 		rot.get_axis_angle(axis, angle);
-		axis.normalize();
-		angular_velocity = axis.normalized() * (angle / p_step);
-
-		motion = new_transform.origin - get_transform().origin;
-		do_motion = true;
-
+		angular_velocity = axis * (angle / p_step);
```

Comment:
Better keep the `axis.normalize();` line to be safe here.

Reply:
I'm a bit suspicious about that part of `get_axis_angle()`:
https://github.com/godotengine/godot/blob/ec3f220098f4bf5a10809e6682a9a24ff93d85cb/core/math/basis.cpp#L891-L932
I have no idea if it's exact math and gives normalized axes in all cases, so I would rather be 100% safe and keep the extra `normalize()` since it's not performance critical code.